src/circuit/circuit_frame.rs

```diff
@@ -2900,24 +2943,33 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
             &g.num_tag,
         )?;
 
+        let is_cons_or_hide =
+            constraints::or(&mut cs.namespace(|| "is cons or hide"), &is_cons, &is_hide)?;
+
+        let is_cons_or_strcons = constraints::or(
+            &mut cs.namespace(|| "is cons or strcons"),
+            &is_cons,
+            &is_strcons,
+        )?;
+
+        let is_cons_or_strcons_or_hide = constraints::or(
+            &mut cs.namespace(|| "is cons or srtcons or hide"),
+            &is_cons_or_hide,
+            &is_strcons,
+        )?;
+
         let res_tag = pick(
             &mut cs.namespace(|| "Op2 result tag"),
-            &is_cons,
+            &is_cons_or_strcons,
             &cons_tag,
```

Comment:
This is confusing to me. The result of a valid `strcons` operation should never be a `Cons`, right? It should be a `Str`.

Reply:
`cons_tag` is defined to be `strcons` or `cons`, depending on the operation.